crates/oxc_parser/src/ts/statement.rs

```diff
@@ -510,4 +512,44 @@ impl<'a> Parser<'a> {
         let expr = self.with_context(Context::Decorator, Self::parse_lhs_expression)?;
         Ok(self.ast.decorator(self.end_span(span), expr))
     }
+
+    pub fn eat_modifiers_before_declaration(&mut self) -> Result<ModifierFlags> {
+        let mut flags = ModifierFlags::empty();
+        let mut modifiers = self.ast.new_vec();
+        while let Ok(modifier_flag) = self.cur_kind().try_into() {
+            let span = self.start_span();
+            // Const can only modify type parameters.
+            if modifier_flag == ModifierFlags::CONST {
+                break;
+            }
+            flags.set(modifier_flag, true);
+            let kind = self.cur_kind();
+            self.bump_any();
+            modifiers.push(Self::modifier(kind, self.end_span(span)));
+        }
+
+        self.state.modifiers = Some(modifiers);
+        Ok(flags)
+    }
+
+    fn modifier(kind: Kind, span: Span) -> Modifier {
+        match kind {
+            Kind::Abstract => Modifier::Abstract(span),
+            Kind::Declare => Modifier::Declare(span),
+            Kind::Private => Modifier::Private(span),
+            Kind::Protected => Modifier::Protected(span),
+            Kind::Public => Modifier::Public(span),
+            Kind::Static => Modifier::Static(span),
+            Kind::Readonly => Modifier::Readonly(span),
+            Kind::Override => Modifier::Override(span),
+            Kind::Async => Modifier::Async(span),
+            Kind::Const => Modifier::Const(span),
+            Kind::In => Modifier::In(span),
+            Kind::Out => Modifier::Out(span),
+            Kind::Export => Modifier::Export(span),
+            Kind::Default => Modifier::Default(span),
+            Kind::Accessor => Modifier::Accessor(span),
+            _ => unreachable!(),
+        }
+    }
```

Comment:
I'm not sure what to do here. Seems this function should belong to the AST builder, but that would require exposing `lexer::Kind` to `oxc_ast`. Any suggestions?